diffblue/cover-client-js
# Preserve package path when writing tests [TG-8226]

### Context/purpose

The client library currently dumps all test files directly into the user specified test directly, but should preserve the package path (e.g. `com/diffblue/javademo/`).

### Implementation details

Use the `sourceFilePath` directory when working out where a test file should be written/may exist.

### QA instructions

Use `Analysis.run` (or `writeTests`) to write test files, you should see them written with the package path preserved .e.g `testDir/com/diffblue/javademo/TicTacToeTest.java`.

### Any unrelated changes?

Fix docs section title.

Disable `no-any` tslint rule in tests, remove disable comments.

### PR Checklist

- [x] I have given the PR a meaningful title which can be understood out of context (used in the changelog)
- [x] I have read through my code and fixed the indentation, followed naming conventions and removed leftover debugging
- [x] I have added the Jira ticket number(s) to the title with the format "Description of the feature [TG-123, TG-456]" and have copied the QA instructions to the Jira ticket (if one exists)
